### PR TITLE
(BKR-1641) Added chmod and modified_at exec commands

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

File: spec/beaker/host/unix/file_spec.rb [hunk 1]
```diff
@@ -196,6 +196,28 @@ def logger
...
+
+      context 'not recursive' do
+        it 'calls execute with chmod' do
+          path = '/path/to/file'
+          mod = '+x'
+
+          expect( instance ).to receive(:execute).with("chmod -R #{mod} #{path}")
+          instance.chmod(mod, path, true)
+        end
+      end
```
Comment (gimmyxd) [inline spec/beaker/host/unix/file_spec.rb:218]: the contexts are switched, first should be `not recursive` and second `recursive` as per test expectation

File: spec/beaker/host/windows/exec_spec.rb [hunk 2]
```diff
@@ -52,6 +58,41 @@ def to_s
       end
     end
 
+    describe '#modified_at' do
+      require 'ostruct'
+      before do
+        allow(instance).to receive(:exec).and_return(OpenStruct.new(stdout: stdout))
+      end
```
Comment (gimmyxd) [inline spec/beaker/host/windows/exec_spec.rb:65]: you can use `double(stdout: stdout)` and no need to require ostruct

File: lib/beaker/host/unix/exec.rb [hunk 3]
```diff
@@ -101,6 +101,16 @@ def touch(file, abs=true)
     (abs ? '/bin/touch' : 'touch') + " #{file}"
   end
 
+  # Update ModifiedDate on a file
+  # @param [String] file Path to the file
+  # @param [String] timestamp Timestamp to set
+  def modified_at(file, timestamp = nil)
+    require 'date'
+    time = timestamp ? DateTime.parse("#{timestamp}") : DateTime.now
```
Comment (gimmyxd) [inline lib/beaker/host/unix/exec.rb:109]: Do we care if timestamp can't be parsed, or we let beaker explode?

File: lib/beaker/host/pswindows/file.rb [hunk 4]
```diff
@@ -19,4 +19,5 @@ def file_exist?(path)
     result = exec(Beaker::Command.new("if exist #{path} echo true"), :acceptable_exit_codes => [0, 1])
     result.stdout =~ /true/
   end
+
```
Comment (gimmyxd) [inline lib/beaker/host/pswindows/file.rb:22]: you can revert this change

**Comment by gimmyxd** [comment]
retest this please

**Comment by gimmyxd** [comment]
retest this please

**Comment by gimmyxd** [comment]
The failures in `acceptance/tests/base.external_resources_test.rb` are unrelated as the fedora epel repository is not accessible anymore, giving 404

**Comment by sebastian-miclea** [comment]
> The failures in `acceptance/tests/base.external_resources_test.rb` are unrelated as the fedora epel repository is not accessible anymore, giving 404

I saw that, I've created a PR to remove EPEL5

File: lib/beaker/host/unix/exec.rb (hunk 3 above)
Comment (sebastian-miclea) [reply lib/beaker/host/unix/exec.rb:109]: Explode sounds so drastic, it's true, it will raise an exception, but that would have happened even with an invalid cli command. 
That's why I used DateTime, it's more flexible when parsing, compared to Time.

Comment (gimmyxd) [reply lib/beaker/host/unix/exec.rb:109]: yes 😅

**Comment by sebastian-miclea** [comment]
Passing build: 
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/

**Comment by gimmyxd** [comment]
Seems like 5.5.13 is not anymore on builds delivery and some tests try to download it 😭

**Comment by highb** [comment]
Jenkins test this please

**Comment by highb** [comment]
> Seems like puppet-agent 5.5.13 is not anymore on builds delivery and some tests try to download it 😭
Yeah, I think I've fixed up that issue in `beaker-puppet`. Hopefully this re-run goes green.

**Comment by gimmyxd** [comment]
@highb this is also neede since we can't test anymore with legacy puppet and osx-1012: https://github.com/puppetlabs/ci-job-configs/pull/6994
